### PR TITLE
Add edit command

### LINKED ISSUE #868: Feature: Add a new  command to simplify editing SDKMAN!'s configuration
- [ x] I have read the [CONTRIBUTING guidelines](CONTRIBUTING.md)

**Feature request**
Changing SDKMAN!'s configuration currently means:

- remembering where the configuration file is located, consulting the online documentation in the worst case
- opening the file in your editor of choice

This feels clunky at best, thus I propose to add an `edit` command which opens the configuration file in the standard editor, or if not configured, falls back to open the file in `vi`.

### PR DESCRIPTION
**Pull request by helpermethod**
Fixes #868.

### CODE REVIEW

**Comment by helpermethod** [comment]
Needs a few more tests but already looks pretty good.

File: src/main/bash/sdkman-edit.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+function __sdk_edit() {
+	local -r editor=${EDITOR:=vi}
+
+	if ! command -v "$editor" > /dev/null; then
+		__sdkman_echo_red $'No default editor configured.\nPlease set the default editor with the EDITOR environment variable.'
```
Comment (marc0der) [inline src/main/bash/sdkman-edit.sh:23]: Can we split this over two lines with a single line opening where the `\n` is? Also, can we make the second line yellow as it is informational?

File: src/test/groovy/sdkman/specs/EditCommandSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+		bash.start()
+		bash.execute("source $bootstrapScript")
+
+		when:
+		setupEnv(bash)
+		bash.execute("sdk edit")
+
+		then:
+		verifyOutput(bash.output, sdkmanBaseDirectory)
```
Comment (marc0der) [inline src/test/groovy/sdkman/specs/EditCommandSpec.groovy:21]: Nice!!!

File: src/main/bash/sdkman-edit.sh (hunk 1 above)
Comment (helpermethod) [reply src/main/bash/sdkman-edit.sh:23]: Good idea!

Comment (helpermethod) [reply src/main/bash/sdkman-edit.sh]: Resolved the issue, but its hard to test on my Mac without a real root account 😞 .

**Comment by marc0der** [comment]
Great work!

**Pull request merged**